### DISCUSSION TITLE
Object has validation errors: origin_country_id (origin_country_id 0 is invalid)

### DISCUSSION THREAD

**Discussion by SrdjanPaunovic** [etsy/open-api#663] [Q&A] [answered]
We are getting errors when uploading files to the listing but the files are successfully uploaded. 
The used endpoint is: `/v3/application/shops/{shopId}/listings/{listingId}/files`. In the response, we are getting this error `Object has validation errors: origin_country_id (origin_country_id 0 is invalid)`.

Can someone help us to understand when this problem occurs? Because it is not always, but still annoys our users.

**Comment by etsyachristensen** [comment]
@SrdjanPaunovic Can you provide some example listing ids? We can't really debug this without this information. The error is happening because something in the listing or it's related information is off. Ideally this shouldn't happen because the files have nothing to do with shipping and country of origin (especially if it's a digital listing). So we will of course look into it. It will just help to have some listing ids where this has occurred.

**Reply by SrdjanPaunovic** [reply]
Here is the listing id: 1348354307. 
It failed at 2022-11-16 18:27:19.000 +0100 if this can help to see in your logs.

**Reply by etsyachristensen** [reply]
Do you by any chance have the response headers saved for this entry? I can find out the exact error if you have the etsy-request-uuid response header value.

**Reply by SrdjanPaunovic** [reply]
We don't have that info for now. Can you find this error by message or by timestamp? Or to see in your code where are you handling this type of validation?

**Reply by etsyachristensen** [reply]
There's a number of possible causes. What I see is that the listing you provided does have a shipping profile id. Granted the profile is the "Digital" listing profile. Normally on digital listings you don't want to have a shipping profile. The challenge is in removing it. I can't really go by the timestamp as we have thousands of logs of all kinds that happen over short time spans. We'll keep looking though.

**Reply by etsyachristensen** [reply]
I believe we found the issue. For the moment, the fastest solution would be to have the client add a billing credit card. At the moment, the shop appears to not have one. This will allow our system to fetch the shop's origin country based on their card. (I don't know WHY our system does it this way.)

We are working on a more permanent resolution of setting that country id based on the country iso we have stored. Hopefully we'll have that out before too long.

**Reply by SrdjanPaunovic** [reply]
We'll ask the user to add his credit card. Thank you very much for your quick response.

**Comment by etsyachristensen** [comment, marked as answer]
@SrdjanPaunovic We have posted a change that will hopefully correct the problem moving forward. If you continue to have issues, please let us know.
